portfolio/src/main/java/com/whz/portfolio/resource/StockAcquiredPublisher.java

```diff
@@ -25,29 +31,44 @@
   private final ConnectionFactory connectionFactory;
 
   StockAcquiredPublisher() {
-    String serviceName = System.getenv("RABBITMQ_SERVICE");
-    exchangeName = System.getenv("RABBITMQ_EXCHANGE");
-    exchangeType = System.getenv("RABBITMQ_EXCHANGE_TYPE");
+    String serviceName = getenv("RABBITMQ_SERVICE", "localhost");
+
+    exchangeName = getenv("RABBITMQ_STOCK_ACQUIRED_EXCHANGE", "StockAcquiredPublisher");
+    exchangeType = getenv("RABBITMQ_STOCK_ACQUIRED_EXCHANGE_TYPE", "fanout");
 
     connectionFactory = new ConnectionFactory();
     connectionFactory.setHost(serviceName);
     logger.debug("Started stock acquired publisher");
   }
 
-  public void send(double data) {
+  /**
+   * Notifies the account about a purchase.
+   *
+   * @param id of the account
+   * @param value - the money spent
+   */
+  public void send(String id, double value) {
     try (final Connection connection = connectionFactory.newConnection();
         final Channel channel = connection.createChannel()) {
       channel.exchangeDeclare(exchangeName, exchangeType);
-      channel.basicPublish(exchangeName, "", null, toBytes(data));
+      StockAcquiredData data = new StockAcquiredData(id, value);
+      String message = JsonSerialization.serialized(data);
+      channel.basicPublish(exchangeName, "", null, message.getBytes());
       logger.debug("Stock acquired publisher sending: " + data);
     } catch (IOException | TimeoutException e) {
       logger.debug(e.getMessage());
```

Comment:
I don't know if this is enough if you want to debug your program. Maybe you change it like this:

```suggestion
      logger.debug("your fancy message", e);
```

I noticed this in a few other places. Look over it again completely.